### PR TITLE
8323066: gc/g1/TestSkipRebuildRemsetPhase.java fails with 'Skipping Remembered Set Rebuild.' missing

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  the test is too sensitive to the amount of allocation, using an arbitrary value of `G1MixedGCLiveThresholdPercent`. This caused random test failures depending on how objects were allocated after GC in old gen, and the amount of allocations during startup.

Using a value of 0 should guarantee that no old region will ever be selected for rebuild, resulting in the correct expected message.

Testing: local testing of test case, gha

Thanks,
  Thomas

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8323066](https://bugs.openjdk.org/browse/JDK-8323066): gc/g1/TestSkipRebuildRemsetPhase.java fails with 'Skipping Remembered Set Rebuild.' missing (**Bug** - P4)

### Reviewers
 * [Albert Mingkun Yang](https://openjdk.org/census#ayang) (@albertnetymk - **Reviewer**) ⚠️ Review applies to [4456a0c9](https://git.openjdk.org/jdk/pull/17309/files/4456a0c9fe4530950c6b6f817faec6375cfb4d0b)
 * [Ivan Walulya](https://openjdk.org/census#iwalulya) (@walulyai - **Reviewer**) ⚠️ Review applies to [4456a0c9](https://git.openjdk.org/jdk/pull/17309/files/4456a0c9fe4530950c6b6f817faec6375cfb4d0b)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17309/head:pull/17309` \
`$ git checkout pull/17309`

Update a local copy of the PR: \
`$ git checkout pull/17309` \
`$ git pull https://git.openjdk.org/jdk.git pull/17309/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17309`

View PR using the GUI difftool: \
`$ git pr show -t 17309`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17309.diff">https://git.openjdk.org/jdk/pull/17309.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17309#issuecomment-1881402963)

### CODE REVIEW

**Review by albertnetymk — APPROVED** [review]

File: test/hotspot/jtreg/gc/g1/TestSkipRebuildRemsetPhase.java [hunk 1]
```diff
@@ -58,24 +58,19 @@ public static void main(String[] args) throws Exception {
...
-        private static Object alloc() {
-            // Since G1MixedGCLiveThresholdPercent is 20%, make sure to allocate object larger than that
-            // so that it will not be collected and the expected message printed.
-            final int objectSize = WhiteBox.getWhiteBox().g1RegionSize() / 3;
-            Object ret = new byte[objectSize];
-            return ret;
-        }
     }
-}
+}
```
Comment (walulyai) [inline test/hotspot/jtreg/gc/g1/TestSkipRebuildRemsetPhase.java:76]: Flagged for no newline at end of file

**Review by walulyai — APPROVED** [review]

**Comment by tschatzl** [comment]
Thanks @albertnetymk @walulyai for your reviews
/integrate

**Pull request closed without merging**